txtorcon/torcontrolprotocol.py

```diff
@@ -102,7 +102,40 @@ def unlisten(self, cb):
         self.callbacks.remove(cb)
 
     def got_update(self, data):
-        # print self.name,"got_update:",data
+        for cb in self.callbacks:
+            cb(data)
+
+class StemEvent(Event):
+    """
+    A class which parses the event information into
+    `stem.response.ControlMessage instance
+    <https://stem.torproject.org/api/response.html#stem.response.ControlMessage>`_
+
+    The information is then parsed into `stem.response.events.Event subclass instance
+    <https://stem.torproject.org/api/response.html#stem.response.events.Event>`_
+    """
+    def got_update(self, data):
+        import stem.response
+        # Add appropriate headers for parsing into stem.response.ControlMessage instance
+        lines = data.split('\n')
+        if len(lines) > 1:
+            response_data = "650-" + self.name + "\r\n"
+            for line in range(len(lines)-1):
+                response_data += "650-" + lines[line] + "\r\n"
+            response_data += "650 " + lines[-1] + "\r\n"
+        else:
+            response_data = "650 " + self.name + " "+ lines[0] + "\r\n"
+        try:
+            response_data = stem.response.ControlMessage.from_str(response_data)
+            stem.response.convert('EVENT', response_data)
```

Comment:
We cannot do something like,
`stem_response = stem.response.convert('EVENT', response_data)`
This is because, when we pass `response_data` into the function, it does not return any value but, does an in-place conversion of the variable